bucket_test.go

```diff
@@ -300,6 +301,73 @@ func TestBucket(t *testing.T) {
 						assert.Error(t, badBucket.Check(ctx))
 					},
 				},
+				{
+					id: "TestSharedCredentialsOption",
+					test: func(t *testing.T, b Bucket) {
+						require.NoError(t, b.Check(ctx))
+
+						newFile, err := os.Create(filepath.Join(tempdir, "creds"))
+						require.NoError(t, err)
+						defer newFile.Close()
+						_, err = newFile.WriteString("[my_profile]\n")
+						require.NoError(t, err)
+						awsKey := fmt.Sprintf("aws_access_key_id = %s\n", os.Getenv("AWS_KEY"))
+						_, err = newFile.WriteString(awsKey)
+						require.NoError(t, err)
+						awsSecret := fmt.Sprintf("aws_secret_access_key = %s\n", os.Getenv("AWS_SECRET"))
+						_, err = newFile.WriteString(awsSecret)
+						require.NoError(t, err)
+
+						sharedCredsOptions := S3Options{
+							SharedCredentialsFilepath: filepath.Join(tempdir, "creds"),
+							SharedCredentialsProfile:  "my_profile",
+							Region:                    s3Region,
+							Name:                      s3BucketName,
+						}
+						sharedCredsBucket, err := NewS3Bucket(sharedCredsOptions)
+						require.NoError(t, err)
+						assert.NoError(t, sharedCredsBucket.Check(ctx))
+					},
+				},
+				{
+					id: "TestSharedCredentialsUsesCorrectDefaultFile",
+					test: func(t *testing.T, b Bucket) {
+						require.NoError(t, b.Check(ctx))
+
+						sharedCredsOptions := S3Options{
+							SharedCredentialsProfile: "default",
+							Region:                   s3Region,
+							Name:                     s3BucketName,
+						}
+						sharedCredsBucket, err := NewS3Bucket(sharedCredsOptions)
+						env := "HOME"
+						if runtime.GOOS == "windows" {
+							env = "USERPROFILE"
+						}
+						fileName := filepath.Join(os.Getenv(env), ".aws/credentials")
+						_, err = os.Stat(fileName)
+						if err == nil {
```

Comment:
There's a helper called "os.IsNotExist" which you should use to look at this error message.

(there's also an os.IsExist, but never use it.)